pageserver/src/restore_local_repo.rs

```diff
@@ -180,6 +286,16 @@ pub fn import_timeline_wal(walpath: &Path, timeline: &dyn Timeline, startpoint:
     let mut segno = startpoint.segment_number(pg_constants::WAL_SEGMENT_SIZE);
     let mut offset = startpoint.segment_offset(pg_constants::WAL_SEGMENT_SIZE);
     let mut last_lsn = startpoint;
+
+    let checkpoint_bytes = timeline.get_page_at_lsn_nowait(ObjectTag::Checkpoint, startpoint)?;
+    let mut checkpoint = CheckPoint::decode(&checkpoint_bytes)?;
+    if checkpoint.nextXid.value == 0 {
```

Comment:
(I think I commented on this before, but: ) How can nextXid ever be zero? I'm pretty sure Postgres initializes it to a valid value.

Reply:
And I have already replied it, but;)
This is actually not a posgres cehckpoint record. It is checkpoint record maintained by pageserver (nextXid, nextMultixacts,...). This check is used to initialize new checkpoint record.
Well, we certainly can change get_page_at_lsn_nowait to return error when page is not found. But right now behavior of this function is that it returns zero page when requested version can not be located. 

As you have mentioned, nextXid can not contain zero value. So this check is used to determine that there was no stored checkpoint in object storage and it has to be initialized. I will add comment explaining it.

Reply:
Does the CheckPoint object exist at all in that case? Or does it only have nextXid as zero?

I think it would be more natural to create the CheckPoint object in import_timeline_from_postgres_datadir, from the control file.

Reply:
> Does the CheckPoint object exist at all in that case?
It doesn't exists and get_page_at_lsn_nowait returns zero page.

Reply:
I think that one extra check in import_timeline_wal is better than adding code for extraction and storing checkpoint record in import_timeline_from_postgres_datadir. But if you think that it is more correct to create checkpoint record in this function, then I will do it.

Reply:
Yeah, I think that would be more clear.